tools/api-metadata-extractor/go.mod

```diff
@@ -0,0 +1,7 @@
+module api-metadata-extractor
+
+go 1.22.5
```

Comment:
nit: should we align with the repo on 1.23.0

ref. https://github.com/flightctl/flightctl/blob/7a7136932bd5785438386538e4fa9f3f329c766a/go.mod#L3